plugins/modules/instance.py

```diff
@@ -1220,6 +1240,19 @@ def _update_instance(self) -> None:
                     "non-updatable field".format(self._instance.label)
                 )
 
+        pg = params.get("placement_group")
+        if pg is not None:
+            if pg.get("id") != self._instance.placement_group.id:
+                self.fail(
+                    "failed to update instance {0}: placement_group.id is a "
+                    "non-updatable field".format(self._instance.label)
+                )
+            if pg.get("compliant_only"):
```

Comment:
@lgarber-akamai The value of `compliant_only` is not returned from API and we don't have a state like tf to find its old value, so I'm thinking returning a warning here to remind user that compliant_only is not updatable.

Reply:
Sounds good to me!